.github/workflows/workflow.ncl

```diff
@@ -0,0 +1,82 @@
+# This is an artisinal, opinionated github workflow contract to complement the
+# auto-generated one. Currently, it's a subset (i.e. a stricter contract) that
+# enforces some more idiomatic nickel conventions (like, less permissiveness
+# with strings vs arrays of strings, less nulls, etc.). Eventually, we may also
+# experiment with fancier things like ADTs.
+#
+# # No "matrix"
+#
+# Our first opinion (that we're testing out here) is lack of support for
+# github's matrix. Let's instead try to generate multiple jobs natively
+# in Nickel. One limitation that I've already encountered: matrix jobs
+# get names like "build-and-test (ubuntu-latest, stable)", and we can't
+# imitate names like these without a matrix because spaces are disallowed
+# in job names.
+let WorkflowExport = import "generated.ncl" in
+{
+  util | not_exported = {
```

Comment:
Is `utils` used in another file, or does it need to be visible? I find it a bit strange to include it as part of the schema as not_exported. Maybe it could go to another file, or the contract could be put in a separate field, or `utils` in a let-binding if it's not used outside of this file?